doc/modules/ROOT/pages/usage/tls.adoc

```diff
@@ -0,0 +1,56 @@
+= TLS (Transport Layer Security) support
+
+The standard way to securely access a nREPL server has been
+to use `ssh` with port forwarding, and letting the nREPL server only bind to `127.0.0.1`.
+
+Sometimes however `ssh` may not be available, and thus you need to
+bind to `0.0.0.0` to make the nREPL server available. The following will explain
+how to do that with TLS.
+
+== Generating certificates and keys
+
+[source,shell]
+----
+$ clojure -Sdeps '{:aliases {:write-cert {:deps {com.github.ivarref/locksmith {:mvn/version "0.1.6"}} :exec-fn com.github.ivarref.locksmith/write-certs!}}}' -T:write-cert :duration-days 365
+Wrote server.keys
+Wrote client.keys
+----
+
+This will generate two 256-bit ECDSA keys. Both files contain the root certificate. Each file also contains the server/client's certificate and private key. The root certificate key is thrown away.
+
+== Starting the nREPL server
+
+[source,clojure]
+----
+=> (require '[nrepl.server :refer [start-server stop-server]])
+nil
+=> (defonce server (start-server :bind "0.0.0.0"
+                                 :port 4001
+                                 :tls? true ; will cause the server to fail if no certs/keys are given
+                                 ; if server.keys is on the server
+                                 :tls-keys-file "server.keys"))
+                                 ; use :tls-keys-str if the contents of server.keys is available as a string,
+                                 ; e.g.
+                                 ; :tls-keys-str (System/getenv "NREPL_SERVER_KEYS")
+='user/server
+----
+
+== Connect with cmdline client
```

Comment:
Might be good to explain here what should 3rd party clients do to support TLS, at least at a high level.